### ISSUE TITLE
Setting `RESIN_HOST_CONFIG_gpu_mem` results in the container (sometimes?) restarting twice

### ISSUE THREAD

**Issue by shaunmulligan** [resin-io/resin-supervisor#108]
Setting a RESIN_HOST_CONFIG_gpu_mem var resulted in weirdness, with the container being started 2 times. The onetime it even seemed that the systemd logs were suddenly attached to the output of the logs... Not sure if its reproducible yet...
```
14.03.16 10:45:53 [+0000] Updating application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:45:53 [+0000] Killing application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:46:42 [+0000] Starting application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:46:48 [+0000] Systemd init system enabled.
14.03.16 10:46:53 [+0000] Started application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:46:55 [+0000] Updating application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:46:55 [+0000] Killing application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:46:57 [+0000] [ OK ] Started LSB: Raise network interfaces..
14.03.16 10:46:58 [+0000] [ OK ] Reached target Network.
14.03.16 10:46:58 [+0000] [ OK ] Reached target System Initialization.
14.03.16 10:46:58 [+0000] [ OK ] Reached target Timers.
14.03.16 10:46:58 [+0000] [ OK ] Reached target Basic System.
14.03.16 10:46:58 [+0000] Starting Resin.io User Application...
14.03.16 10:46:58 [+0000] [ OK ] Started Resin.io User Application.
14.03.16 10:46:58 [+0000] Starting /etc/rc.local Compatibility...
14.03.16 10:46:58 [+0000] Starting Permit User Sessions...
14.03.16 10:46:58 [+0000] PID returned by clone(): 150
14.03.16 10:46:58 [+0000] child_func(): PID = 1
14.03.16 10:46:58 [+0000] child_func(): PPID = 0
14.03.16 10:46:58 [+0000] [ OK ] Started /etc/rc.local Compatibility.
14.03.16 10:46:58 [+0000] [ OK ] Started Permit User Sessions.
14.03.16 10:46:58 [+0000] [ OK ] Reached target Multi-User System.
14.03.16 10:46:58 [+0000] [ OK ] Reached target Graphical Interface.
14.03.16 10:46:58 [+0000] Starting Update UTMP about System Runlevel Changes...
14.03.16 10:46:58 [+0000] [ OK ] Started Update UTMP about System Runlevel Changes.
14.03.16 10:47:05 [+0000]
14.03.16 10:47:10 [+0000] Killed application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:47:11 [+0000] Installing application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:47:17 [+0000] Installed application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:47:17 [+0000] Starting application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:47:19 [+0000] Started application 'registry.resin.io/eglexper/a9aba9b03a8590504cb223d64150010057bbebf3'
14.03.16 10:47:20 [+0000] Systemd init system enabled.
14.03.16 10:47:22 [+0000] PID returned by clone(): 139
14.03.16 10:47:22 [+0000] child_func(): PID = 1
14.03.16 10:47:22 [+0000] child_func(): PPID = 0
14.03.16 10:47:22 [+0000] hello_triangle.bin: triangle.c:174: init_ogl: Assertion `state->surface != ((EGLSurface)0)' failed.
```

**Comment by pcarranzav** [comment]
I think I know where this may come from. The device applies the OS changes, then reboots, and doesn't get to save the RESIN_HOST env vars to the DB, so it picks it up as an update after the reboot. I'll chase this and incorporate it while implementing #107 
Thanks!

**Comment by shaunmulligan** [comment]
Awesome :+1:

**Comment by pcarranzav** [comment]
Another (even nastier) symptom of the same problem: changing a variable that was already set will cause a reboot loop. It can be worked around by deleting the variable before setting the new value. Fix will also come with #107.
